test/integration/helpers.js

```diff
@@ -358,6 +359,18 @@ function createSubprocess(args, done, opts = {}) {
     });
   });
 
+  /**
+   * Emulate node's exit code for fatal signal. Allows tests to see the same
+   * exit code as the mocha cli.
+   */
+  mocha.on('exit', (code, signal) => {
+    if (signal) {
+      mocha.exitCode =
+        SIGNAL_OFFSET +
+        (typeof signal == 'string' ? os.constants.signals[signal] : signal);
+    }
+  });
```

Comment:
Just noting for future reference: I was confused at first that `exitCode` isn't newly referenced elsewhere. It's some built-in process plumbing. A couple of integration tests fail if you comment this out locally.

Reply:
Yeah, this was a bit puzzling, but for mocha self-testing IIRC it was necessary to mimic the way node's exit codes work (which is already posix compliant). I'll resolve this thread, but if you want additional detail here in addition to what is already in the doc block above, please let me know.